### PR TITLE
Add `isBlockInMainChain` endpoint

### PR DESCRIPTION
**Pull request by Lbqds**

### CODE REVIEW

File: app/src/main/scala/org/alephium/app/ServerUtils.scala [hunk 1]
```diff
@@ -334,6 +334,16 @@ class ServerUtils(implicit
         .map(failedInIO)
     } yield BlockEntry.from(block, height)
 
+  def isBlockInMainChain(blockFlow: BlockFlow, blockHash: BlockHash): Boolean = {
```
Comment (polarker) [inline app/src/main/scala/org/alephium/app/ServerUtils.scala:337]: We'd better return `Try[Boolean]`. `Try` wraps IO error

File: app/src/main/scala/org/alephium/app/ServerUtils.scala [hunk 2]
```diff
@@ -334,6 +334,16 @@ class ServerUtils(implicit
...
+    val result = for {
+      height <- blockFlow.getHeight(blockHash)
```
Comment (tdroxler) [inline app/src/main/scala/org/alephium/app/ServerUtils.scala:339]: should we return a `NotFound` if the hash doesn't exist?

Comment (Lbqds) [reply app/src/main/scala/org/alephium/app/ServerUtils.scala]: Now it will return an error to be consistent with other endpoints.

**Review by tdroxler — APPROVED** [review]
LGTM

**Review by polarker — APPROVED** [review]
LGTM

**Pull request merged**